### DISCUSSION TITLE
Enhancement ideas for Projects, TOC, Zotero: academic paper writing workflow

### DISCUSSION THREAD

**Discussion by do-not-perceive-me** [Zettlr/Zettlr#3644] [Ideas]
I recently downloaded Zettlr to write my literature review and thesis in and have some notes for things I feel could be useful for an academic writing workflow. 

context: I'm currently using Zotero for references and pdf annotations, Notion to write literature notes and general project planning, and Zettlr to write the full paper. I'm also starting to move some of my concept/topics pages from Notion to Zettlr. 

### A. Better way to rearrange files within a project

Originally, I had a file for each chapter, and rearranged my topics with the file but I would prefer shorter files to mimic the zettlekasten method. Now I have a file for each main topic, and the order might change as I write the final document. I also find it useful to use smaller topic files because I can set a word count goal for each section. 

I would like an easier way to move the order of the files around within the Project Directory. Right now you can only sort by name and date. Currently I'm changing the order by numbering the filenames but that's not ideal with a larger number of files, or if you're changing the structure more frequently. 

**Proposal 1:** have a Table of Contents that covers the whole Project, which would allow you to rearrange the file and sections in that TOC.  Having a project TOC would be a great feature to see what the final output TOC would look like anyways. 

**Proposal 2:** have the option to define an outline file (similar to the [§§etherpad.md example described by Hendrik](https://www.hendrik-erz.de/post/how-i-work-part-v-zettlr-and-academic-markdown)). The docs mentioned that this could leave files, but maybe it possible to auto generate a directory page with all the file within the project already listed, that can then be moved around. 

[[intro]]
[[body-intro]]     where this starts with h1
  [[topic A]]       and these files might start with h2, etc
  [[topic X]]
[[conclusion]]

**Proposal 3:** ability to remove sort on the file directory so that you can customize the file order. 

I also tried setting the file order with glob pattern, but that doesn't change the order. 

### B. Table of Contents improvements

- the ability to select and copy and paste the TOC. It's not possible to highlight and select it right now. I would like to be able to do this so that I could paste it into a To-Do list, mindmap software, or Notion etc. Right now, I would need to export it with a TOC, then copy it over, and that's not ideal. 
- folding/collapse option
- vertical lines to better see the nesting structure
- task list within the TOC. I write by outline, and I write all my headings first. When I'm done writing a section, I would like to be able to mark the heading as complete, collapse it, and then have it more greyed out in the TOC. Right now I'm marking unfinished ones with a symbol and removing when I'm done. I tried to use css to change the color of that heading in the TOC if it contains that symbol, but I don't think that's possible with css

### C. see all comments in a section in sidebar

I write comments to myself to remind myself of things as I'm writing, It would be nice if the sidebar could have a tab to show all the comments, with the ability to jump to that part of the file. 

### D. File directory, word count & targets

- setting or calculating the total word count target for a project. Usually you would have a word count goal for a whole paper, so it would be useful to see if I've divided the writing targets among the files sufficiently to reach the overall goal. 
- ability to mark a file as complete, and change opacity. I think it would also be good to have the option to change the appearance of files that are marked complete or in-active. Or any other way to  favorite a file or de-emphasize or hide files. I just want to see the files that I'm currently working on more easily, without putting them in a different folder, because that messes up the order for project export. It's useful to have tags that mark as complete, but it's still a lot of visual clutter. 

### E. notes integration with Zotero 

I think this would be more complicated, but having the citation or reference link give a link to the Zotero item would be useful. I'd also like a better way to export PDF annotations from Zotero into md for zettlr, that somehow links to the reference. II'm using mdnotes to export the annotations, but don't know how I would relate that file to the citation. My idea would be that if you insert a citation with @author2000, that you could also relate it to a file [[author2000]], so that every time you insert the citation, a related notes file is also linked.

discussed [here as well ](https://github.com/Zettlr/Zettlr/discussions/2937)

### F. Better explanation for Word Docx export on the docs

My main reason for using Zettlr is some of my supervisors prefer Word

 I had to really struggle to figure out how to format everything and export it nicely, but I finally managed. I can probably help write some documentation for this, now that I've figured it out, but I know very little about github or how to contribute to the documentation

I think a simple change would be to include `reference-doc: reference.docx` in the defaults file, and add the docx file to the folder with the tutorial that is there on download. Or add a note saying to put your reference docx in the same folder and change the name there.

**Comment by jorg03** [comment]
Hi there,

I also struggle with the Word export and how format everything. I would appreciate any help. E.g. is there any possibility to include a path to a reference docx. And if not how do I change something like the font's color in the title?
Thanks

**Reply by do-not-perceive-me** [reply]
1. Yes you can include the path to the ref docx, in the defaults file in assets manager. 
2. You change the font color using the Styles pane in Word

I'm in the process of editing the Zettlr docs to include instructions for Word export, you can see what I've written here in the [pull request](https://github.com/Zettlr/zettlr-docs/pull/155/files?short_path=aea9e29#diff-aea9e292c26ddb9cbc1130b84eef065b6454fff53e45990e52356eac55dba259). It explains the answers to your 2 questions a bit more as well.

**Reply by jorg03** [reply]
Thanks a lot, this worked!

**Comment by derWalter** [comment]
this proposal should get more attention 👍

**Comment by carljoseph** [comment]
A year later ... this proposal rules!

For me, I'd love a more Scrivener type sorting option where I can drag drop the files and folders into a particular order live on the screen. Couple that with a multi-select and you can read a whole section of your thesis on the one screen/tab to get a better sense of flow, continuity errors, etc. This is one of the huge beneficial features of Scrivener and I'd love to see it in Zettlr.
